Automattic/wp-calypso
# Me: add data layer and Redux plumbing for site block listing

Although it is currently possible to block specific sites in Reader so they're not shown or recommended to you any more, there's nowhere in Calypso to see a complete listing of your current blocks, or remove them.

This PR adds the data layer and Redux plumbing to consume blocks from the new listing endpoint `/me/blocks/sites` (D10891-code).

### Testing instructions

* Check that the following test suites pass:

`npm run test-client client/state/data-layer/wpcom/me/blocks`
`npm run test-client client/state/reader/site-blocks/`
